receiver/prometheusreceiver/config_test.go

```diff
@@ -161,3 +161,147 @@ func TestRejectUnsupportedPrometheusFeatures(t *testing.T) {
 	require.Equal(t, wantErrMsg, gotErrMsg)
 
 }
+
+func TestNonExistentAuthCredentialsFile(t *testing.T) {
+	factories, err := componenttest.NopFactories()
+	assert.NoError(t, err)
+
+	factory := NewFactory()
+	factories.Receivers[typeStr] = factory
+	cfg, err := configtest.LoadConfig(path.Join(".", "testdata", "invalid-config-prometheus-non-existent-auth-credentials-file.yaml"), factories)
+	require.NoError(t, err)
+	require.NotNil(t, cfg)
+	err = cfg.Validate()
+	require.NotNil(t, err, "Expected a non-nil error")
+
+	wantErrMsg := `receiver "prometheus" has invalid configuration: error checking authorization credentials file "/nonexistentauthcredentialsfile" - stat /nonexistentauthcredentialsfile: no such file or directory`
+
+	gotErrMsg := err.Error()
+	require.Equal(t, wantErrMsg, gotErrMsg)
+}
+
+func TestTLSConfigCertFileExists(t *testing.T) {
```

Comment:
Nit: This is testing what happens when the cert file is non-existent, the title of the test makes me think it's testing the opposite.